docs/templates/pallet.adoc

```diff
@@ -0,0 +1,40 @@
+:source-highlighter: highlight.js
+:highlightjs-languages: rust
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+= Pallet Name
+
+== Purpose
+
+This is a freeform description of the tasks that this pallet fulfills
+
+== Glossary
+
+* _Term_ -- definition of the term
+
+== Config
+
+* `ConfigType` -- desription of config. Include the possible values if there is a set of them.
+
+== Dispatchables
+
+[[dispatchable_name]]
+=== `++dispatchable_name++` link:https://google.com[{github-icon},role=heading-link]
+```rust
+pub fn dispatchable_name(
+    param1: Type1,
+    param2: Type2
+) -> DispatchResult
+```
+Freeform description of the dispatchable. It is good to include the important things that should be included there.
+
+**Params:**
+
+* `param1: Type1` -- description of the parameter
+
+**Errors:**
+
+* `ErrorName` -- description of conditions, when this error happens
+
+**Events:**
+
+* `EventName(param1, param2)` -- description of event, if needed (they are often self-explanatory)
```

Comment:
I suggest we add another field for advanced details, or any further niche clarifications. I'm not sure on the naming for this header.

Yet, I think it's beneficial to have this on the template, due to:
- the location for these advanced stuff will be determined
- in the future, probably we will just follow the headers of this template. And we might forget about adding these niche details, since they were not mentioned in the template.
- I'm also ok having this _"advanced details"_ header commented out, since, maybe some pallets won't require the explanation for advanced details. Still, I think the header should be there, in one way or another

Reply:
I agree with you. Such section would be useful.
I will think of a formatting for it

Reply:
I have added important info and more editing